governance/xc_admin/packages/xc_admin_frontend/components/tabs/General.tsx

```diff
@@ -287,306 +303,317 @@ const General = ({ proposerServerUrl }: { proposerServerUrl: string }) => {
   }
 
   const handleSendProposalButtonClick = async () => {
-    if (pythProgramClient && dataChanges && !isMultisigLoading) {
-      const instructions: TransactionInstruction[] = []
-      const publisherInPriceStoreInitializationsVerified: PublicKey[] = []
-
-      for (const symbol of Object.keys(dataChanges)) {
-        const multisigAuthority = readOnlySquads.getAuthorityPDA(
-          PRICE_FEED_MULTISIG[getMultisigCluster(cluster)],
-          1
-        )
-        const fundingAccount = isRemote
-          ? mapKey(multisigAuthority)
-          : multisigAuthority
-
-        const initPublisherInPriceStore = async (publisherKey: PublicKey) => {
-          // Ignore this step if Price Store is not initialized (or not deployed)
-          if (!connection || !(await isPriceStoreInitialized(connection))) {
-            return
-          }
+    const handleSendProposalButtonClickAsync = async () => {
```

Comment:
this is just a copy paste refactor so we catch the promise error

Reply:
You should remove the `async` marker from the outer function since it's no longer async.

Also, I generally recommend hoisting this kind of thing to the module scope and passing in arguments.  For one, it avoids the perf overhead of creating closures at runtime.  For two, it makes the code much easier to follow and understand.  For three, explicitly passing arguments rather than relying on closure scoped variables makes the information flow much more obvious.

This function is already gigantic and really hard to trace and understand its flow, breaking it up outside the component would help a lot.

Reply:
removed the async keywork, but won't refactor since it's too much work